### PR TITLE
Resize Nodes

### PR DESCRIPTION
**Pull request by OliverEGreen**
The new node design has been upated to take up less space on the graph. 
This PR is mostly adjustments made to height, paddings and margins.
A new, smaller node context menu icon has been switched out for the old one.
Each node's context menu can now display a column of checkboxes, if IsCheckable is set to true. 
The original right/left click functionality has been restored to nodes at the zoomed-out level (right click opens context menu, left clicks selects and drags).
The main adjustment made to the nodes is that the PresentationGrid now lives centrally, and occupies a single layout column on the NodeView. This mostly affects nodes like the Watch and Watch3D nodes. These also now have an updated resize control.

This PR addresses issues:
[DYN-4014] https://jira.autodesk.com/projects/DYN/issues/DYN-4014
[DYN-3961] https://jira.autodesk.com/projects/DYN/issues/DYN-3961
[DYN-4032] https://jira.autodesk.com/projects/DYN/issues/DYN-4032
[DYN-3936] https://jira.autodesk.com/projects/DYN/issues/DYN-3936
[DYN-3937] https://jira.autodesk.com/projects/DYN/issues/DYN-3937

### CODE REVIEW

**Comment by QilongTang** [comment]
Please check regression `DynamoCoreWpfTests.PythonNodeCustomizationTests.PythonNodeHasLabelDisplayingCurrentEngine`

File: src/Libraries/PythonNodeModelsWpf/PythonNode.cs [hunk 1]
```diff
@@ -73,8 +73,14 @@ public void CustomizeView(PythonNode nodeModel, NodeView nodeView)
...
-            nodeView.PresentationGrid.Visibility = Visibility.Visible;
-            nodeView.PresentationGrid.Children.Add(new EngineLabel(nodeModel));
+            EngineLabel engineLabel = new EngineLabel(nodeModel);
+            Canvas.SetZIndex(engineLabel, 5);
+            engineLabel.HorizontalAlignment = HorizontalAlignment.Left;
+            engineLabel.VerticalAlignment = VerticalAlignment.Bottom;
+            engineLabel.Margin = new Thickness(14, -4, 0, 4);
+            nodeView.grid.Children.Add(engineLabel);
+            Grid.SetColumn(engineLabel, 0);
+            Grid.SetRow(engineLabel, 3);
```
Comment (QilongTang) [inline src/Libraries/PythonNodeModelsWpf/PythonNode.cs:83]: I guess these changes could be related to the regression I mentioned in comment

Comment (OliverEGreen) [reply src/Libraries/PythonNodeModelsWpf/PythonNode.cs:83]: Hi Aaron, 

yes, I moved this label to the node's 'grid' away from the 'PresentationGrid'.
Have just updated to fix this test.

**Review by QilongTang — APPROVED** [review]
@OliverEGreen Thanks! LGTM and will merge once all checks passed

**Comment by QilongTang** [comment]
Self CI passed at https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/3584/

**Pull request merged**